### ISSUE TITLE
Embedding The Amazon Connect CCP in Angular

### ISSUE THREAD

**Issue by nabeelahm2** [amazon-connect/amazon-connect-streams#492]
I am trying to embed the CCP in my angular app. I have installed the `npm install amazon-connect-streams` and added  `import "amazon-connect-streams"` statement in my .ts component. However, when I run the application,  I get the following error in my browser console .

```
Uncaught TypeError: Cannot set property 2011-06-15 of [object Object] which has only a getter
    at Object.push../node_modules/amazon-connect-streams/release/connect-streams.js.105.../apis/connect-2017-02-15.min (connect-streams.js:21097)
    at o (connect-streams.js:4)
    at r (connect-streams.js:4)
    at Object../node_modules/amazon-connect-streams/release/connect-streams.js (connect-streams.js:4)
    at __webpack_require__ (bootstrap:84)
    at Module../src/app/core/components/login/login.component.ts (login.component.ts:1)
    at __webpack_require__ (bootstrap:84)
    at Module../src/app/app-routing.module.ts (app-routing.module.ts:1)
    at __webpack_require__ (bootstrap:84)
    at Module../src/app/app.module.ts (app.module.ts:1) 
```

App angular version is 9.1.12 and typescript version is 3.8.3.  Any suggestion how to fix this issue.

**Comment by DanielHuelse** [comment]
I also get this error with Angular 13 and Typescript 4.4. There exists an issue for AWS SDK [Issue 3555](https://github.com/aws/aws-sdk-js/issues/3555) that could be related to this. It would be great if this could be fixed somehow.

**Comment by ctwomblyamzn** [comment]
It looks like this is coming from this line: https://github.com/amazon-connect/amazon-connect-streams/blob/master/src/aws-client.js#L21091. Therefore, I believe this is connected to the AWS SDK issue mentioned above: [Issue 3555](https://github.com/aws/aws-sdk-js/issues/3555). The file aws-client, which I linked earlier is an auto-generated file from the sdk team, so ideally this will be fixed by them. I am going to bump the issue and see if we can get some traction.

**Comment by DanielHuelse** [comment]
Is there any progress on this issue? When can we expect a fix?

**Comment by jonatandmapser** [comment]
Is there any progress on this issue?
in my case it is something strange, since in a previous project it worked correctly, and now when trying to create a new project, it does not work

**Comment by ctwomblyamzn** [comment]
Hi, sorry to hear that. We want to address this from the SDK-side, as this will break every time we update our client if not. I am hoping to get some attention to [this PR soon](https://github.com/aws/aws-sdk-js/issues/3555).

**Comment by DanielHuelse** [comment]
@ctwomblyamzn is there any progress regarding this issue?

**Comment by SeanGroff** [comment]
@ctwomblyamzn [this PR](https://github.com/aws/aws-sdk-js/issues/3555) has been open for 2 years...
Can someone share how to fix this within the amazon-connect-streams code? Even if that means having to re-apply the fix each time the file is generated.

**Comment by DanielHuelse** [comment]
The only way I see to solve the issue is that the Amazon Connect team provides a customized version of the AWS SDK where the changes from the [PR](https://github.com/aws/aws-sdk-js/issues/3555) already applied.

**Comment by SeanGroff** [comment]
> The only way I see to solve the issue is that the Amazon Connect team provides a customized version of the AWS SDK where the changes from the [PR](https://github.com/aws/aws-sdk-js/issues/3555) already applied.

Is this something we can explore since the PR has been open for 2 years now with no sign of it being addressed in the SDK

**Comment by DanielHuelse** [comment]
Since the SDK is hard-bundled we just need to provide a new aws-client.js

What i tried so far is, that i extracted the diff from the PR as patch file:
[strict-mode.txt](https://github.com/amazon-connect/amazon-connect-streams/files/8654331/strict-mode.txt) and applied it manually via `git apply strict-mode.txt` to a local copy of aws-sdk-js.
Then i run `node ./dist-tools/browser-builder.js connect > aws-client.js` to export only the connect client.
So far i only tested it with code example that was mentioned in the PR, but this seemed to work.

I haven't had much time to test it with the amazon-connect-streams project. 

Also I am a bit unsure which version of the AWS SDK is the right version to export to. The current aws-client.js in master states it is v2.553.0, but there were recently changes on the file and the version is still the same.

**Comment by DanielHuelse** [comment]
Since the SDK is hard-bundled we just need to provide a new aws-client.js

What i tried so far is, that i extracted the diff from the PR as patch file:
[strict-mode.txt](https://github.com/amazon-connect/amazon-connect-streams/files/8654331/strict-mode.txt) and applied it manually  via `git apply strict-mode.txt` to local copy of aws-sdk-js
then i run `node ./dist-tools/browser-builder.js connect > aws-client.js` to export only the connect client.
So far i only tested it with code example that was mentioned in the PR, but this seemed to work.

I haven't had time to test it with the amazon-connect-streams project. 

Also I am a bit unsure which version of the AWS SDK is the right version to export to. The current aws-client.js in master states it is v2.553.0, but there were recently changes on the file and the version is still the same. So it is hard to tell which is the right version.

**Comment by DanielHuelse** [comment]
I looked into the matter of patching the SDK somehow, but from my investigation it seems like that the amazon-connect-streams library operates with an internal version of the AWS SDK v2 - so I see no chance to provide an PR.

**Comment by CGarces** [comment]
Fixed at #4166
@nabeelahm2 can you please close this issue?

**Comment by seiyako** [comment]
An AWS SDK bug that was fixed on v2.3.0
